pkg/machineproviders/providers/openshift/machine/v1beta1/failuredomain/failuredomain.go

```diff
@@ -339,7 +339,42 @@ func convertAWSResourceReference(subnet machinev1beta1.AWSResourceReference) *ma
 		return subnetv1
 	}
 
-	return subnetv1
+	return nil
+}
+
+// ConvertAWSResourceReferenceToBeta1 creates machinev1beta1.awsResourceReference from machinev1.awsResourceReference.
+func ConvertAWSResourceReferenceToBeta1(subnet *machinev1.AWSResourceReference) machinev1beta1.AWSResourceReference {
```

Comment:
Being pedantic, but this should be named
```suggestion
// ConvertAWSResourceReferenceV1ToV1Beta1 creates a machinev1beta1.awsResourceReference from machinev1.awsResourceReference.
func ConvertAWSResourceReferenceV1ToV1Beta1(subnet *machinev1.AWSResourceReference) machinev1beta1.AWSResourceReference {
```